boards/DCM/Inc/App/states/App_SharedStates.h

```diff
@@ -0,0 +1,126 @@
+    const struct DcmCanRxInterface *can_rx_interface)
+{
+    return App_CanRx_FSM_BRAKE_GetSignal_BRAKE_IS_ACTUATED(can_rx_interface) ==
+           CANMSGS_FSM_BRAKE_BRAKE_IS_ACTUATED_TRUE_CHOICE;
+}
+
+/**
+ * Check if the start switch is pulled up to the on position
+ * @param can_rx_interface The CAN Rx interface to get the CAN signals from
+ * @return true if the start switch is on, false otherwise
+ */
+static inline bool App_SharedStates_IsStartSwitchOn(
+    const struct DcmCanRxInterface *can_rx_interface)
+{
+    return App_CanRx_DIM_SWITCHES_GetSignal_START_SWITCH(can_rx_interface) ==
+           CANMSGS_DIM_SWITCHES_START_SWITCH_ON_CHOICE;
+}
+
+/**
+ * Check if the start switch was pulled up
+ * @param prev_start_switch_position Start switch position of previous tick
+ * @param current_start_switch_position Start switch position of current tick
+ * @return true if the start switch was pulled up, false otherwise
+ */
+static inline bool App_SharedStates_WasStartSwitchPulledUp(
+    bool prev_start_switch_position,
+    bool current_start_switch_position)
+{
+    return !prev_start_switch_position && current_start_switch_position;
+}
+
+/**
+ * Check if the start switch is off
+ * @param can_rx_interface The CAN Rx interface to get the CAN signals from
+ * @return true if the start switch is off, false otherwise
+ */
+static inline bool App_SharedStates_IsStartSwitchOff(
+    const struct DcmCanRxInterface *can_rx_interface)
+{
+    return App_CanRx_DIM_SWITCHES_GetSignal_START_SWITCH(can_rx_interface) ==
+           CANMSGS_DIM_SWITCHES_START_SWITCH_OFF_CHOICE;
+}
+
+/**
+ * Check if both inverters are OK
+ * @param can_rx_interface The CAN Rx interface to get the CAN signals from
+ * @return true if both inverters are OK, false otherwise
+ */
+static inline bool App_SharedStates_AreInvertersOK(
+    const struct DcmCanRxInterface *can_rx_interface)
+{
+    return App_CanRx_INVL_INTERNAL_STATES_GetSignal_D1_VSM_STATE_INVL(
+               can_rx_interface) !=
+               CANMSGS_INVL_INTERNAL_STATES_D1_VSM_STATE_INVL_BLINK_FAULT_CODE_STATE_CHOICE &&
+           App_CanRx_INVR_INTERNAL_STATES_GetSignal_D1_VSM_STATE_INVR(
+               can_rx_interface) !=
+               CANMSGS_INVR_INTERNAL_STATES_D1_VSM_STATE_INVR_BLINK_FAULT_CODE_STATE_CHOICE;
+}
+
+/**
+ * Check if vehicle is over the regen threshold defined by vehicle wheel speed
+ * (EV.4.1.3)
+ * @param can_rx_interface The CAN Rx interface to get the CAN signals from
+ * @return true if the vehicle is over the regen threshold, false otherwise
+ */
+static inline bool App_SharedStates_IsVehicleOverRegenThresh(
+    const struct DcmCanRxInterface *can_rx_interface)
+{
+    return (App_CanRx_FSM_WHEEL_SPEED_SENSOR_GetSignal_LEFT_WHEEL_SPEED(
+                can_rx_interface) > REGEN_WHEEL_SPEED_THRESHOLD_KPH) &&
+           (App_CanRx_FSM_WHEEL_SPEED_SENSOR_GetSignal_RIGHT_WHEEL_SPEED(
+                can_rx_interface) > REGEN_WHEEL_SPEED_THRESHOLD_KPH);
+}
```

Comment:
if this is only used in a single .c file we don't need to place them in the header

Reply:
i think there are a couple more fxns that dont need to be exposed here make sure u go through and check where theyre currently being used

Reply:
done